iroh-net/src/relay/client.rs

```diff
@@ -28,8 +28,8 @@ use tracing::{debug, error, event, info_span, trace, warn, Instrument, Level};
 use url::Url;
 
 use conn::{
-    Conn as RelayClient, ConnBuilder as RelayClientBuilder, ConnReader,
-    ConnReceiver as RelayClientReceiver, ConnWriter, ReceivedMessage,
+    Conn as RelayConn, ConnBuilder as RelayConnBuilder, ConnReader,
+    ConnReceiver as RelayConnReceiver, ConnWriter, ReceivedMessage,
```

Comment:
Wild suggestion, but what if we import `conn:: self as relay` and just use `relay::ConnReceiver`, `relay::Conn` and `relay::ConnBuilder`. 

It's not shorter, but maybe it's a bit more aligned with this idea of stop using prefixes and prefer module references.

What are your thoughts?

Reply:
I originally thought this was a great idea, but actually doing it having `conn as relay` inside `iroh_net::relay::client` was just a bit too weird.

But why bother renaming at all?  Looking at it again I think it is at least as clear to use `Conn` as `RelayConn` given that we're in `iroh_net::relay::client`.  So that's what I did together with improving the docs on `Conn`, `ConnBuilder` and `ConnReceiver`.

What do you think?